src/map/pc.c

```diff
@@ -11405,81 +11460,11 @@ int pc_readdb(void) {
 	FILE *fp;
 	char line[24000],*p;
 
-	//reset
-	memset(pc->exp_table,0,sizeof(pc->exp_table));
-	memset(pc->max_level,0,sizeof(pc->max_level));
-
-	sprintf(line, "%s/"DBPATH"exp.txt", map->db_path);
-
-	fp=fopen(line, "r");
-	if(fp==NULL){
-		ShowError("can't read %s\n", line);
-		return 1;
-	}
-	while(fgets(line, sizeof(line), fp)) {
-		int jobs[CLASS_COUNT], job_count, job, job_id;
-		int type;
-		int maxlv;
-		char *split[4];
-		if(line[0]=='/' && line[1]=='/')
-			continue;
-		if (pc_split_str(line,split,4) < 4)
-			continue;
+	/**
+	 * Read and load into memory, the exp_group_db.conf file.
+	 */
+	pc->read_exp_db();
```

Comment:
Please note that `pc_readdb()` can be called multiple times (it's called by the atcommand `@reloadpcdb`).

In order to avoid memory leaks or dangling pointers, it might now be necessary to clean up before calling `pc->read_exp_db()` (see below the call to `pc->clear_skill_tree()` before `pc->read_skill_tree()` for example).